code/modules/reagents/chemistry/reagents/pyrotechnic_reagents.dm

```diff
@@ -245,6 +245,37 @@
 			L.extract_cooldown = max(0, L.extract_cooldown - 20)
 	..()
 
+/datum/reagent/teslium/energized_jelly/energized_ooze
+	name = "Energized Ooze"
+	description = "Electrically-charged Ooze. Boosts Oozeling's nervous system, but only shocks other lifeforms."
+	reagent_state = LIQUID
+	color = "#CAFF43"
+	taste_description = "slime"
+	overdose_threshold = 30
+
+/datum/reagent/teslium/energized_jelly/energized_ooze/on_mob_life(mob/living/carbon/M)
+	if(isjellyperson(M))
+		shock_timer = 0 //immune to shocks
+		M.AdjustAllImmobility(-40, FALSE)
+		M.adjustStaminaLoss(-2, 0)
+		if(isluminescent(M))
+			var/mob/living/carbon/human/H = M
+			var/datum/species/jelly/luminescent/L = H.dna.species
+			L.extract_cooldown = max(0, L.extract_cooldown - 20)
+	if(isoozeling(M))
+		shock_timer = 0 //immune to shocks
+		M.AdjustAllImmobility(-40, FALSE)
+		M.adjustStaminaLoss(-2, 0)
+	..()
+
+/datum/reagent/teslium/energized_jelly/overdose_process(mob/living/carbon/M)
+	shock_timer++
+	if(isoozeling(M) || isjellyperson(M))
+		shock_timer = (rand(0,2))
```

Comment:
`shock_timer` here has no meaning; `on_mob_life` happens no matter if the mob is overdosing or not, so `shock_timer` is always neutralized there. You check for overdose at `on_mob_life` or do something else.

Reply:
Ngl I mostly copy-pasted the teslium code for this

/datum/reagent/teslium/on_mob_life(mob/living/carbon/M)
	shock_timer++
	if(shock_timer >= rand(5,30)) //Random shocks are wildly unpredictable
		shock_timer = 0
		M.electrocute_act(rand(5,20), "Teslium in their body", 1, 1) //Override because it's caused from INSIDE of you
		playsound(M, "sparks", 50, 1)
	..()

are both `shock_timer`'s not needed or?

Reply:
Let's get this simple. `shock_timer` is *only* for `on_mob_life`. For `overdose_process`, you should devise another way to shock reasonably. Just copypasta a bit and use `overdose_shock_timer` instead. Don't forget to declare `overdose_shock_timer`. You also should override at `/datum/reagent/teslium/energized_jelly/energized_ooze` since `/datum/reagent/teslium/energized_jelly` has no overdose threshold, therefore, can't overdose. (I missed that)

Reply:
ahhh, I thought that as the `energized_jelly/energized_ooze` has an `overdose_threshold` that when it gets an OD it would use energized_jelly's `overdose_process`
![image](https://user-images.githubusercontent.com/17776299/104674472-a1f5d000-56db-11eb-908a-47aff1a275e1.png)
![image](https://user-images.githubusercontent.com/17776299/104674829-33654200-56dc-11eb-87e6-a5af48109c9b.png)

Reply:
I'm being dumb here, sent you a msg on discord